plutus-extra/src/Plutus/Contract/Test/Extra.hs

```diff
@@ -97,3 +148,122 @@ walletFundsChangeWithAccumStateImpl exact contract inst wallet toDlt =
                  , indent 2 (viaShow w)
                  ]
       pure result
+
+{- | Check that the funds at a computed address meet some condition.
+ The address is computed using data acquired from contract's writer instance.
+
+  @since 1.0.1.0
+-}
+valueAtAddressComputedFromAccumState ::
+  forall
+    (w :: Type)
+    (s :: Row Type)
+    (e :: Type)
+    (a :: Type)
+    (contract :: Type -> Row Type -> Type -> Type -> Type).
+  ( Monoid w
+  , IsContract contract
+  ) =>
+  contract w s e a ->
+  ContractInstanceTag ->
+  (w -> Maybe Ledger.Address) ->
+  (Ledger.Value -> Bool) ->
+  TracePredicate
+valueAtAddressComputedFromAccumState contract inst addressGetter check =
+  utxoAtAddressComputedFromAccumState contract inst addressGetter $ \addr utxoMap -> do
+    let value = foldMap (Ledger.txOutValue . Ledger.txOutTxOut) utxoMap
+        result = check value
+    unless result $
+      tell @(Doc Void) ("Funds at address" <+> pretty addr <+> "were" <> pretty value)
+    return result
+
+{- | Check that the datum at a computed address meet some condition.
+ The address is computed using data acquired from contract's writer instance.
+
+  @since 1.0.1.0
+-}
+dataAtAddressComputedFromAccumState ::
+  forall
+    (datum :: Type)
+    (w :: Type)
+    (s :: Row Type)
+    (e :: Type)
+    (a :: Type)
+    (contract :: Type -> Row Type -> Type -> Type -> Type).
+  ( FromData datum
+  , Monoid w
+  , IsContract contract
+  ) =>
+  contract w s e a ->
+  ContractInstanceTag ->
+  (w -> Maybe Ledger.Address) ->
+  (datum -> Bool) ->
+  TracePredicate
+dataAtAddressComputedFromAccumState contract inst addressGetter check =
+  utxoAtAddressComputedFromAccumState contract inst addressGetter $ \addr utxoMap -> do
+    let datums = mapMaybe (uncurry $ getTxOutDatum @datum) $ Map.toList utxoMap
+        result = any check datums
+    unless result $
+      tell @(Doc Void)
+        ( "Data at address" <+> pretty addr <+> "was"
+            <+> foldMap (foldMap pretty . Ledger.txData . Ledger.txOutTxTx) utxoMap
+        )
+    return result
+
+{- | Extract UTxOs at a computed address and call continuation returning
+ Boolean value based on both the address and UTxOs.
+ The address is computed using data acquired from contract's writer instance.
+
+  @since 1.0.1.0
+-}
+utxoAtAddressComputedFromAccumState ::
```

Comment:
Is it possible to name this more concisely? This name is awkward and _very_ long, as are a bunch of others in this module.

Reply:
Changed those just to `utxoAtComputedAddress`, hopefully it would be better. I haven't touched the previous function names as changing them would be API-breaking, but can do if necessary.